### DISCUSSION TITLE
Greenfield API - Payment type field

### DISCUSSION THREAD

**Discussion by nemanja-z** [btcpayserver/btcpayserver#2394] [Technical Support]
Getting an information about payment type(on-chain and off-chain) requires fetching information from a server for every invoice. It would be great if a user could get that information by calling https://docs.btcpayserver.org/api/v1/stores/{storeId}/invoices for all invoices at once, instead of calling https://docs.btcpayserver.org/api/v1/stores/{storeId}/invoices/{invoiceId}/payment-methods
for every invoice.

**Comment by bolatovumar** [comment]
Sounds reasonable. @Kukks any reasons for why this might not be a good idea?

**Comment by NicolasDorier** [comment]
https://github.com/btcpayserver/btcpayserver/pull/6757

**Reply by bollerdominik** [reply]
Any updates on this? This is needed for our integration to work
